guppylang/compiler/expr_compiler.py

```diff
@@ -492,11 +492,8 @@ def visit_DesugaredArrayComp(self, node: DesugaredArrayComp) -> Wire:
         assert isinstance(array_ty, OpaqueType)
         array_var = Variable(next(tmp_vars), array_ty, node)
         count_var = Variable(next(tmp_vars), int_type(), node)
-        hugr_elt_ty = (
-            ht.Option(node.elt_ty.to_hugr())
-            if node.elt_ty.linear
-            else node.elt_ty.to_hugr()
-        )
+        # See https://github.com/CQCL/guppylang/issues/629
```

Comment:
not sure about the relevance of this link?

Reply:
This is to explain why we use `Option` for both linear and classical arrays